File: testdata/java_sources/Application.java
package com.example;

/**
 * Main application entry point.
 */
public class Application {

    public static void main(String[] args) {
        Application app = new Application();
        app.run();
    }

    public void run() {
        NativeBridge bridge = new NativeBridge();
        int result = bridge.compute(21);
        System.out.println("Result: " + result);
    }

    /**
     * Unused helper - dead code.
     */
    private void printUsage() {
        System.out.println("Usage: application");
    }

    /**
     * Unused shutdown hook - dead code.
     */
    protected void onShutdown() {
        // cleanup
    }
}

File: testdata/kotlin_sources/Application.kt
package com.example.kotlin

/**
 * Kotlin application entry point.
 */
fun main() {
    val app = Application()
    app.run()
}

class Application {
    fun run() {
        val result = computeWrapper(21)
        println("Result: $result")
    }

    /**
     * Unused shutdown method - dead code.
     */
    fun shutdown() {
        println("shutting down")
    }

    /**
     * Unused configuration loader - dead code.
     */
    private fun loadConfig(): Map<String, String> {
        return mapOf("host" to "localhost", "port" to "8080")
    }
}
